packages/typescript/src/index.js

```diff
@@ -58,6 +65,58 @@ module.exports = {
       },
     ],
 
+    // Recommended rules that require type information
+    '@typescript-eslint/no-unsafe-argument': 'off',
+    '@typescript-eslint/no-unsafe-assignment': 'off',
+    '@typescript-eslint/no-unsafe-call': 'off',
+    '@typescript-eslint/no-unsafe-member-access': 'off',
+    '@typescript-eslint/no-unsafe-return': 'off',
+    '@typescript-eslint/restrict-template-expressions': 'off',
+
+    // Our rules that require type information
+    '@typescript-eslint/consistent-type-exports': 'error',
+    '@typescript-eslint/naming-convention': [
+      'error',
+      {
+        selector: 'default',
+        format: ['camelCase'],
+        leadingUnderscore: 'forbid',
+        trailingUnderscore: 'forbid',
+      },
+      {
+        selector: 'enumMember',
+        format: ['PascalCase'],
+      },
+      {
+        selector: 'variable',
+        format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
+      },
+      {
+        selector: 'typeLike',
+        format: ['PascalCase'],
+      },
+      {
+        selector: 'interface',
+        format: ['PascalCase'],
+        custom: {
+          regex: '^I[A-Z]',
+          match: false,
+        },
+      },
+    ],
+    '@typescript-eslint/no-confusing-void-expression': 'error',
+    '@typescript-eslint/no-meaningless-void-operator': 'error',
+    '@typescript-eslint/no-unnecessary-boolean-literal-compare': ' error',
+    '@typescript-eslint/no-unnecessary-qualifier': 'error',
+    '@typescript-eslint/no-unnecessary-type-arguments': 'error',
+    '@typescript-eslint/prefer-includes': 'error',
+    '@typescript-eslint/prefer-nullish-coalescing': 'error',
+    '@typescript-eslint/prefer-readonly': 'error',
```

Comment:
In classes with variables that are never mutated, this makes sure we define it as `readonly`. Not sure if we want to use this.

Reply:
Hmm. I'm OK with trying this. Not entirely sure how I feel about it yet.

Reply:
Start with `warn`?

Reply:
We could :thinking: 

The problem with warnings is that since they don't get highlighted by CI, people don't notice them and they build up over time. That's why we haven't used warnings for anything here so far.

Reply:
I believe there is ways for warnings to show inline in GitHub, but in general I don't think we should use warnings. This rule is auto-fixable, so it's easy to implement regardless.